lib/gias_api_client.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require "zip"
+
+class GiasApiClient
+  def get_files
+    data = fetch_extract
+    temp_files_from_zipped_response(data)
+  end
+
+private
+
+  def fetch_extract
+    client = Savon.client(wsdl: Rails.application.config.gias_api_schema)
+    res = client.call(
+      :get_extract,
+      message: { "tns:Id" => Rails.application.config.gias_extract_id },
+      wsse_auth: [Rails.application.config.gias_api_user, Rails.application.config.gias_api_password],
+      wsse_timestamp: true,
+    )
+    parts = parse_body(res.http)
+    parts.second.body.to_s
+  end
+
+  def temp_files_from_zipped_response(data)
+    files = {}
+
+    # We need to write the zip file to disk rather than read it from a buffer because of this:
+    # https://github.com/rubyzip/rubyzip#notice-about-zipinputstream
+    zipfile = Tempfile.new(encoding: "ASCII-8BIT")
+    zipfile.write(data.force_encoding("ASCII-8BIT"))
```

Comment:
I doubt we need this line if we have already specified the encoding in zipfile = Tempfile.new(encoding: "ASCII-8BIT")

Reply:
This is incredibly touchy about encodings. If you're happy with it, I'd like to leave this as it is, as it definitely works like this